app/javascript/controllers/flash_controller.js

```diff
@@ -0,0 +1,19 @@
+import { Controller } from "stimulus";
+import { flashNotice, flashError } from "../shopify_app/flash_messages";
+
+export default class extends Controller {
+  connect() {
+    if (!window.app) {
+      return;
+    }
+    const flashData = JSON.parse(this.element.dataset.flash);
+
+    if (flashData.notice) {
+      flashNotice(flashData.notice);
+    }
+
+    if (flashData.error) {
+      flashError(flashData.error);
```

Comment:
Rails has convinience helpers for `notice` and `alert` flashes. I think it will be better to stick with the same flash names.
```suggestion
    if (flashData.alert) {
      flashError(flashData.alert);
```